integration/command.go

```diff
@@ -157,7 +157,7 @@ var cleanupInspektorGadget *command = &command{
 var cleanupSPO *command = &command{
 	name: "RemoveSecurityProfilesOperator",
 	cmd: `
-	kubectl delete seccompprofile -n security-profiles-operator --all
+	kubectl delete seccompprofile --all --all-namespaces
```

Comment:
Yeah, it will. But I would say that it's exactly the reason why we provide `-no-deploy-spo`. Don't you think?

Reply:
We could anyway add an additional check to only remove the SPO if it was deployed by ourselves. It shouldn't be difficult to do.

Reply:
> Don't you think?

If a user wants to run the integration tests they probably will use `make integration-tests` without paying that much attention to all possible flags. The default behavior is to try to deploy SPO. Maybe we can change it to **not deploy SPO** and skip tests depending on it? 

 > We could anyway add an additional check to only remove the SPO if it was deployed by ourselves. It shouldn't be difficult to do.

Is it really easy to implement? How can we check that SPO was deployed by ourselves and that the user wasn't running it before?

Reply:
> Maybe we can change it to not deploy SPO and skip tests depending on it?
>
Makes sense. In the end, most gadgets don't depend on it. 

> Is it really easy to implement? How can we check that SPO was deployed by ourselves and that the user wasn't running it before?
>
You are right. It is not that obvious now that I think better about it. I thought we could check the creation time of the SPO daemon, but it is too heuristic, it could have been confidentially just deployed:

```bash
kubectl get SecurityProfilesOperatorDaemon -n security-profiles-operator spod -o jsonpath={.metadata.creationTimestamp}
```

Maybe annotating it would be more accurate (Of course, after ensuring it has no implications):

```bash
kubectl annotate -n security-profiles-operator SecurityProfilesOperatorDaemon/spod context=inspektor-gadget-tests
```

Reply:
How about requiring `-cleanup` flag for deleting such components. We can pass this explicitly in CI.

Reply:
> Maybe annotating it would be more accurate (Of course, after ensuring it has no implications):

But how can you make the annotation conditional to not annotate an existing installation? 

> How about requiring -cleanup flag for deleting such components. We can pass this explicitly in CI.

The problem with this approach is that it'll leave resources on the cluster after running the integration tests. 

I'm wondering if there is a programmatic way to check if the security profiles operator is installed, then we can skip / run the tests without caring about who installed SPO.

Reply:
One way I can think is checking for SPO crds. A very simple check but might work for us?

Reply:
Yes, even something simple like `kubectl get seccompprofile -A` and checking it return a list instead of error should be fine.

Reply:
I opened the [issue](https://github.com/kinvolk/inspektor-gadget/issues/976) to track this discussion. I will go ahead and merge this change.